sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt

```diff
@@ -0,0 +1,59 @@
+package com.mapbox.maps
+
+import android.content.Context
+
+/**
+ * Defines configuration [MapInitOptions] for a [MapboxMap]. These options can be used when adding a
+ * map to your application programmatically (as opposed to via XML). If you are using a MapFragment,
+ * you can pass these options in using the static factory method newInstance(MapboxMapOptions).
+ * If you are using a [MapView], you can pass these options in using the constructor
+ * MapView(Context, MapboxMapOptions). If you add a map using XML, then you can apply these options
+ * using custom XML tags.
+ *
+ * @property context the context of the application.
+ * @property resourceOptions Resource options when using a MapView. Access token required when using a Mapbox service. Please see [https://www.mapbox.com/help/create-api-access-token/](https://www.mapbox.com/help/create-api-access-token/) to learn how to create one.More information in this guide [https://www.mapbox.com/help/first-steps-android-sdk/#access-tokens](https://www.mapbox.com/help/first-steps-android-sdk/#access-tokens).
+ * @property mapOptions Describes the map options value when using a MapView.
+ */
+data class MapInitOptions constructor(
+  val context: Context,
+  var resourceOptions: ResourceOptions = getDefaultResourceOptions(context),
+  var mapOptions: MapOptions = getDefaultMapOptions(context)
```

Comment:
I think we also need pixel ratio here, so that it can overwrite the one from context

Reply:
Do you mean make pixelRatio as a param for MapInitOptions?

Reply:
yeah, something like that, we are also parsing the pixel ratio from the xml, right? It would not be obvious if the user can set it manually

Reply:
That's right. But we must make a clear doc to tell users it's only the default value, will not take effect if they use a custom MapOptions instance here.

Reply:
thinking again, I think user could still specify the pixel ratio manually by building the MapOptions themselves.. @kiryldz do you have thoughts here?

Reply:
@pengdev why would we actually need to override pixel ratio from context as part of default options? I know we have issues if accessing pixel ratio from static instances in case of multi-display but what's the use-case when pixle ratio taken from concrete context is not correct?

Reply:
Another potential problem I see is if we override it here and pass to core but will rely from pixel ratio from context in some other methods.

Reply:
@kiryldz 
> @pengdev why would we actually need to override pixel ratio from context as part of default options? I know we have issues if accessing pixel ratio from static instances in case of multi-display but what's the use-case when pixle ratio taken from concrete context is not correct?

The pixel ratio is configurable, it doesn't have to be the context's default density. But on the other hand, I think it can be configured through xml and when users build the ResourceOptions themselves. I think we can leave the current API as it is for now.

> Another potential problem I see is if we override it here and pass to core but will rely from pixel ratio from context in some other methods.

The pixel ratio set in the MapInitOptions#MapOptions is saved in the MapboxMap object and Style object, and we make sure it is used across our APIs instead of using the pixel ratio from the context.